content/posts/0004-writing-buildpacks.md

```diff
@@ -0,0 +1,97 @@
+Maybe you’ve heard some chatter about Cloud Native Buildpacks, or at least the value in containerizing your applications. Buildpacks provide a seamless mechanism for doing just that. There are a few implementations of the buildpack concept available, but the general takeaways can be summarized as follows:
+
+> “Buildpacks provide a seamless experience from source code to a running container” 
+
+> “Dockerfiles aren’t the only option anymore” 
+
+> “Everything is open source and community driven”
+
+This is all very well and good for making containerized applications in a pinch, but what happens when I need a specific buildpack for my use case that doesn’t exist yet? As a community-driven project, is the onus on me to create that buildpack myself? Where does one even begin?
+
+The Paketo Buildpacks project strives to make answering those questions transparent, approachable, and simple.  If you haven’t been introduced to Paketo Buildpacks before, check out this [blog post](https://medium.com/paketo-buildpacks/building-apps-for-kubernetes-get-to-know-paketo-buildpacks-6dc29b0f3cf3) for a full rundown. The idea of being a community-driven project is splashed across our [website](paketo.io), and [blog posts](https://medium.com/paketo-buildpacks/2021-paketo-buildpacks-roadmap-8f3977833b32) for a reason. As an engineer who recently learned the ropes of working on buildpacks myself, I’ll explain why building a buildpack isn’t as hard as you might think, and clue you in on some tips and tricks I've learned along the way.
+
+
+## Creating a Paketo-style buildpack
+
+There are definitely some need-to-know basics when it comes to buildpacks that have been well documented by the Cloud Native Buildpacks project in [the docs](https://buildpacks.io/docs/concepts/). Major concepts such as the buildpack lifecycle phases, and the other components that work alongside buildpacks are explained there. Once you have the basics down, there is a [simple tutorial](https://buildpacks.io/docs/buildpack-author-guide/create-buildpack/) to create a buildpack in Bash.
+
+
+### Packit is here to help!
+
+Kicking things up a notch, the [Creating a Paketo Buildpack tutorial](https://paketo.io/docs/tutorials/create-paketo-buildpack/) on the [Paketo website](paketo.io) outlines how to create a really simple buildpack in **Golang**. With Golang, it’s possible to implement more complex logic with less work than with Bash. A shining star of the tutorial is the introduction of the [Packit library](github.com/paketo-buildpacks/packit), which simplifies the repetitive parts of writing a buildpack, such as [dependency installation](https://github.com/paketo-buildpacks/packit/blob/main/postal) and [file system manipulation](https://github.com/paketo-buildpacks/packit/blob/main/fs). As a part of the Cloud Native Buildpacks project, our buildpacks follow the [CNB Spec](https://github.com/buildpacks/spec). A huge benefit of Packit is that it helps abide by the CNB spec without too much effort.
```

Comment:
Maybe move the first couple sentences here into the intro paragraph before so it flows better into Packit section. So something like "...create a buildpack in Bash. A buildpack written in Bash works great for a simple buildpack, but the Paketo contributors have adopted Golang as our language of choice to..."